functorch/experimental/output_dtype.py

```diff
@@ -0,0 +1,136 @@
+from functools import partial 
+
+import torch
+import torch.utils._pytree as pytree
+from torch.fx.experimental.proxy_tensor import (
+    ProxyTorchDispatchMode,
+    disable_proxy_modes_tracing,
+    ProxyTorchDispatchMode,
+    track_tensor_tree,
+    unwrap_proxy,
+)
+from torch.utils._python_dispatch import (
+    _get_current_dispatch_mode,
+    _pop_mode_temporarily,
+)
+from torch._C import DispatchKey
+from torch._functorch.eager_transforms import (
+    _unwrap_all_tensors_from_functional,
+    _wrap_all_tensors_to_functional,
+)
+from torch._ops import HigherOrderOperator
+from torch._subclasses.fake_tensor import FakeTensorMode
+
+
+output_dtype = HigherOrderOperator("output_dtype")
+output_dtype.fallthrough(DispatchKey.PythonDispatcher)
+output_dtype.fallthrough(DispatchKey.PythonTLSSnapshot)
+output_dtype.fallthrough(DispatchKey.ADInplaceOrView)
+output_dtype.fallthrough(DispatchKey.BackendSelect)
+output_dtype.fallthrough(DispatchKey.AutocastCPU)
+
+
+def trace_output_dtype(proxy_mode, func_overload, op, out_dtype, *args):
+    if not isinstance(op, torch._ops.OpOverload):
+        raise ValueError("output_dtype's first argument must be an OpOverload")
+
+    with disable_proxy_modes_tracing():
+        casted_args = pytree.tree_map_only(
+            torch.Tensor, lambda arg: arg.to(dtype=out_dtype), args
+        )
+        out = op(*casted_args)
+
+    node_args = (op, out_dtype, *args)
+    proxy_args = pytree.tree_map(partial(unwrap_proxy, proxy_mode), node_args)
+    out_proxy = proxy_mode.tracer.create_proxy(
+        "call_function", func_overload, proxy_args, {}, name="output_dtype"
+    )
+    return track_tensor_tree(out, out_proxy, constant=None, tracer=proxy_mode.tracer)
```

Comment:
what does this do?

Reply:
It's like a wrapper to have make_fx keep track of the results of running this function.